src/worker-thread/dom/elementSibling.ts

```diff
@@ -0,0 +1,33 @@
+import { NodeType } from '../../transfer/TransferrableNodes';
+import { Element } from './Element';
+import { Node } from './Node';
+
+export function getPreviousElementSibling(node: Node): Element | null {
+  let parentNodes = node.parentNode && node.parentNode.childNodes;
+  if (!parentNodes) {
+    return null;
+  }
+
+  for (let i = parentNodes.indexOf(node) - 1; i >= 0; i--) {
+    let node = parentNodes[i];
+    if (node.nodeType === NodeType.ELEMENT_NODE) {
+      return node as Element;
+    }
+  }
+  return null;
+}
+
+export function getNextElementSibling(node: Node): Element | null {
+  let parentNodes = node.parentNode && node.parentNode.childNodes;
+  if (!parentNodes) {
+    return null;
+  }
+
+  for (let i = parentNodes.indexOf(node) + 1; i < parentNodes.length; i++) {
+    let node = parentNodes[i];
+    if (node.nodeType === NodeType.ELEMENT_NODE) {
+      return node as Element;
+    }
+  }
+  return null;
+}
```

Comment:
Not sure if/how much bundle size matters for work-dom, but this could save some by sharing the logic (at the cost of some readability)

```suggestion
function getElemementSibling(startNode: Node, step: 1 | -1): Element | null {
  let parentNodes = startNode.parentNode?.childNodes;
  if (!parentNodes) {
    return null;
  }

  let i = parentNodes.indexOf(startNode) + step;
  let node;
  while (node = parentNodes[i]) {
    if (node.nodeType === NodeType.ELEMENT_NODE) {
      return node as Element;
    }
    i += step;
  }
  return null;
}

export function getPreviousElementSibling(node: Node): Element | null {
  return getElemementSibling(node, -1);
}

export function getNextElementSibling(node: Node): Element | null {
  return getElemementSibling(node, 1);
}
```

Reply:
code size is most definitely a concern. but there are much bigger fish to fry IMO before we hurt readability